### PR TITLE
actions: Document tests

### PR DESCRIPTION
**Pull request by ghost**
**Related issues**

https://github.com/Open-Telecoms-Data/standard-development-handbook/pull/24#discussion_r1052811423

### CODE REVIEW

File: tests/schema.py [hunk 1]
```diff
@@ -6,6 +6,12 @@
 
 
 class CurrentVersionOFDSSchema(OFDSSchema):
+    """
+    A Schema object for libcoveofds that is used in these tests.
+    The normal Schema object always fetches the latest schema from the web.
```
Comment (lgs85) [inline tests/schema.py]:
```suggestion
    The OFDSSchema object always fetches the latest schema from the web.
```

I assume that this is what's meant here?

File: tests/test_examples_json.py [hunk 2]
```diff
@@ -16,24 +20,30 @@ def get_data(self):
         return {}
 
     def test_additional_fields(self):
+        """Test there is no additional fields. Uses libcoveofds library."""
```
Comment (lgs85) [inline tests/test_examples_json.py]:
```suggestion
        """Test that there are no additional fields. Uses libcoveofds library."""
```

File: tests/test_examples_json.py [hunk 3]
```diff
@@ -16,24 +20,30 @@ def get_data(self):
...
         worker = AdditionalFields(self.schema)
         results = worker.process(self.get_data())
         assert results == {}
 
     def test_jsonschema_validation(self):
+        """Test there is no JSON Schema validation errors. Uses libcoveofds library."""
```
Comment (lgs85) [inline tests/test_examples_json.py]:
```suggestion
        """Test that there are no JSON Schema validation errors. Uses libcoveofds library."""
```

File: tests/test_examples_json.py [hunk 4]
```diff
@@ -16,24 +20,30 @@ def get_data(self):
...
         worker = JSONSchemaValidator(self.schema)
         results = worker.validate(self.get_data())
         # We call .json() so that if there any any problems, the output error messages are usefull.
         results = [r.json() for r in results]
         assert results == []
 
     def test_python_validation(self):
+        """Test there is no Python validation errors. Uses libcoveofds library."""
```
Comment (lgs85) [inline tests/test_examples_json.py]:
```suggestion
        """Test that there are no Python validation errors. Uses libcoveofds library."""
```

File: tests/test_examples_json.py [hunk 5]
```diff
@@ -16,24 +20,30 @@ def get_data(self):
...
         worker = PythonValidate(self.schema)
         results = worker.validate(self.get_data())
         assert results == []
 
 
-class TestNetworkPackage(BaseTest):
+class TestNetworkPackage(BaseTestExampleDataValid):
+    """
+    Tests examples/json/network-package.json is valid.
```
Comment (lgs85) [inline tests/test_examples_json.py]:
```suggestion
    Tests whether examples/json/network-package.json is valid.
```

File: tests/test_examples_json.py [hunk 6]
```diff
@@ -42,7 +52,11 @@ def get_data(self):
             return json.load(fp)
 
 
-class TestMultipleNetworks(BaseTest):
+class TestMultipleNetworks(BaseTestExampleDataValid):
+    """
+    Tests examples/json/multiple-networks.json is valid.
```
Comment (lgs85) [inline tests/test_examples_json.py]:
```suggestion
    Tests examples/json/multiple-networks.json is valid.
```
```suggestion
    Tests whether examples/json/multiple-networks.json is valid.
```

File: tests/test_examples_json.py [hunk 7]
```diff
@@ -42,7 +52,11 @@ def get_data(self):
...
+    See
```
Comment (lgs85) [inline tests/test_examples_json.py]: Is there something else to be added here?

File: tests/test_json.py [hunk 8]
```diff
@@ -17,12 +17,16 @@
 
 
 def test_empty():
+    """Tests files (JSON and other files) are not empty."""
```
Comment (lgs85) [inline tests/test_json.py]:
```suggestion
    """Tests that files (JSON and other files) are not empty."""
```

File: tests/test_json.py [hunk 9]
```diff
@@ -17,12 +17,16 @@
...
     empty_files_paths = [path for path in get_empty_files() if "src/" not in path[0]]
     warn_and_assert(empty_files_paths, "{0} is empty, run: rm {0}", "Files are empty. See warnings below.")
 
 
 def test_indent():
-    """Note this test can often fail on problems that are not to do with indents."""
+    """
+    Test JSON files are indented properly.
```
Comment (lgs85) [inline tests/test_json.py]:
```suggestion
    Test that JSON files are indented properly.
```

File: tests/test_json.py [hunk 10]
```diff
@@ -32,6 +36,7 @@ def test_indent():
 
 
 def test_invalid_json():
+    """Test all JSON files can be parsed."""
```
Comment (lgs85) [inline tests/test_json.py]:
```suggestion
    """Test whether all JSON files can be parsed."""
```

**Review by lgs85 — CHANGES_REQUESTED** [review]
Looks good. Some overly picky copy edits, then happy to merge.

File: tests/schema.py (hunk 1 above)
Comment (ghost) [reply tests/schema.py]: Yes

File: tests/test_examples_json.py (hunk 7 above)
Comment (ghost) [reply tests/test_examples_json.py]: Can't remember. Removed "See"

**Review by lgs85 — APPROVED** [review]

**Pull request merged**